crashtracker/src/receiver.rs

```diff
@@ -32,17 +34,38 @@ pub fn get_unix_socket(socket_path: impl AsRef<str>) -> anyhow::Result<UnixListe
     Ok(unix_listener)
 }
 
-pub fn reciever_entry_point_unix_socket(socket_path: impl AsRef<str>) -> anyhow::Result<()> {
-    let listener = get_unix_socket(socket_path)?;
-    let (unix_stream, _) = listener.accept()?;
-    let stream = BufReader::new(unix_stream);
-    receiver_entry_point(stream)
+pub fn receiver_entry_point_unix_socket(socket_path: impl AsRef<str>) -> anyhow::Result<()> {
+    let rt = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()?;
+    rt.block_on(async_receiver_entry_point_unix_socket(socket_path, true))?;
+    Ok(())
     // Dropping the stream closes it, allowing the collector to exit if it was waiting.
 }
 
 pub fn receiver_entry_point_stdin() -> anyhow::Result<()> {
-    let stream = std::io::stdin().lock();
-    receiver_entry_point(stream)
+    let stream = BufReader::new(tokio::io::stdin());
+    let rt = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()?;
+    rt.block_on(receiver_entry_point(stream))?;
+    Ok(())
+}
+
+pub async fn async_receiver_entry_point_unix_socket(
```

Comment:
do we have a test for this

Reply:
It's tested by the "bin_tests" as both sync and async "entry_point_unix_socket" use this code path